### PR TITLE
slice and cache_range_requests: allow header override

### PR DESCRIPTION
**Pull request by traeak**
This adds the ability to override the internal slice skip header (was hard coded to X-Slicer-Info) and the cache_range_requests ims header (was hard coded to X-Crr-Ims).

### CODE REVIEW

**Comment by bryancall** [comment]
@elsloo Can you please review this?

File: plugins/cache_range_requests/cache_range_requests.cc [hunk 1]
```diff
@@ -111,24 +112,29 @@ create_pluginconfig(int argc, char *const argv[])
...
     }
 
     switch (opt) {
-    case 'p': {
-      DEBUG_LOG("Plugin modifies parent selection key");
-      pc->ps_mode = PS_CACHEKEY_URL;
-    } break;
     case 'c': {
-      DEBUG_LOG("Plugin considers the '%.*s' header", (int)X_IMS_HEADER.size(), X_IMS_HEADER.data());
+      DEBUG_LOG("Plugin considers the CRR ims header");
```
Comment (elsloo) [inline plugins/cache_range_requests/cache_range_requests.cc:122]: Can this be reverted such that the default header is printed out when a custom header is not in use? Specifically, print the value of `DefaultImsHeader` here.

File: plugins/experimental/slice/slice.h [hunk 2]
```diff
@@ -32,8 +32,6 @@
 #define PLUGIN_NAME "slice"
 #endif
 
-constexpr std::string_view X_CRR_IMS_HEADER = {"X-Crr-Ims"};
```
Comment (elsloo) [inline plugins/experimental/slice/slice.h:35]: This constant and all references to `X_CRR_IMS_HEADER` were removed, but `SLICER_MIME_FIELD_INFO` still exists in `HttpHeader.h` (line 39). Consequently there are two areas that still reference the old default `X-Slicer-Info` header name using the constant, and it is injected into requests despite having a "custom" header configuration.

File: plugins/experimental/slice/client.cc [hunk 3]
```diff
@@ -74,7 +75,7 @@ handle_client_req(TSCont contp, TSEvent event, Data *const data)
         data->m_statustype = TS_HTTP_STATUS_REQUESTED_RANGE_NOT_SATISFIABLE;
 
         // First block will give Content-Length
-        rangebe = Range(0, data->m_config->m_blockbytes);
+        rangebe = Range(0, conf->m_blockbytes);
       }
     } else {
       DEBUG_LOG("%p Full content request", data);
```
Comment (elsloo) [inline plugins/experimental/slice/client.cc:81]: Please mitigate the reference to `SLICER_MIME_FIELD_INFO` on line 84 below.

File: plugins/experimental/slice/server.cc [hunk 4]
```diff
@@ -268,8 +268,8 @@ logSliceError(char const *const message, Data const *const data, HttpHeader cons
   crange.toStringClosed(normstr, &normlen);
```
Comment (elsloo) [inline plugins/experimental/slice/server.cc:268]: Please mitigate the reference to `SLICER_MIME_FIELD_INFO` on line 262 above.

**Review by elsloo — CHANGES_REQUESTED** [review]
Please remove the `SLICER_MIME_FIELD_INFO` constant and update the two dependent lines of code to use the new member variables.

File: plugins/cache_range_requests/cache_range_requests.cc (hunk 1 above)
Comment (traeak) [reply plugins/cache_range_requests/cache_range_requests.cc:122]: The other 2 messages in the same function should cover it (based on override being used or not):

DEBUG_LOG("Plugin uses custom CRR ims header: %s", optarg);
DEBUG_LOG("Plugin uses default ims header: %s", pc->ims_header.c_str())

File: plugins/experimental/slice/client.cc (hunk 3 above)
Comment (traeak) [reply plugins/experimental/slice/client.cc:81]: stupid missed this.

File: plugins/experimental/slice/server.cc (hunk 4 above)
Comment (traeak) [reply plugins/experimental/slice/server.cc:268]: missed this sry

**Comment by ezelkow1** [comment]
[approve ci rocky]

**Review by elsloo — APPROVED** [review]
I think this looks good now. Thanks for addressing the feedback @traeak!

**Comment by zwoop** [comment]
This was back ported to 9.2.x in #8898.

**Pull request merged**